main.tf

```diff
@@ -1 +1,98 @@
+# To have a NSG
+data "azurerm_network_security_group" "this" {
```

Comment:
You don't need an extra datasource for this. Because only the NSGs resource ID is needed you can ask for that in variables.tf and depending on if that variable is given associate the NSG or not. Basically same as the windows VM module: https://github.com/qbeyond/terraform-azurerm-windows-vm/blob/00a26a51f4d9f6a48377e87897a5f30e99ecb21e/main.tf#L36

Reply:
Deleted, solved as suggestion.